From 7fb11608e9a1cd9a5471289994865d8875052b55 Mon Sep 17 00:00:00 2001
From: Vladimir Barinov <vladimir.barinov@cogentembedded.com>
Date: Tue, 1 Sep 2015 22:10:49 +0300
Subject: [PATCH] tty: serial: sh-sci: fix kernel oops

Fix kernel oops on shutdown/reboot:
kernel BUG at drivers/tty/serial/sh-sci.c:1565!

Signed-off-by: Vladimir Barinov <vladimir.barinov@cogentembedded.com>
---
 drivers/tty/serial/sh-sci.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/tty/serial/sh-sci.c b/drivers/tty/serial/sh-sci.c
index 2b2e9d4..51eca733 100644
--- a/drivers/tty/serial/sh-sci.c
+++ b/drivers/tty/serial/sh-sci.c
@@ -1903,6 +1903,9 @@ static void sci_shutdown(struct uart_port *port)
 	sci_stop_tx(port);
 	spin_unlock_irqrestore(&port->lock, flags);
 
+	cancel_work_sync(&s->work_tx);
+	cancel_work_sync(&s->work_rx);
+
 	serial_port_out(port, SCSCR, 0x00);
 
 	sci_free_dma(port);
-- 
1.9.1

